internal/metrics/otelmetrics/factory.go

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) 2024 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package otelmetrics
+
+import (
+	"context"
+	"log"
+
+	"github.com/jaegertracing/jaeger/pkg/metrics"
+	"go.opentelemetry.io/otel/attribute"
+	metric "go.opentelemetry.io/otel/metric"
+)
+
+type otelFactory struct {
+	meter metric.Meter
+}
+
+func NewFactory(meterProvider metric.MeterProvider) metrics.Factory {
+	return &otelFactory{
+		meter: meterProvider.Meter("jaeger-v2"),
+	}
+}
+
+func (f *otelFactory) Counter(opts metrics.Options) metrics.Counter {
+	counter, err := f.meter.Int64Counter(opts.Name)
```

Comment:
note: you will also need to include the Namespace string from the factory. We often sub-scope the factories for components via `f1 := f2.Namespace({Name: "mycomponent"})`. You can see how other factories handle that.

Reply:
Noted

Reply:
PTAL, I think normalizing won't have any effect because . / - are acceptable characters in the Naming convention by OTEL.